### DISCUSSION TITLE
Using relative path in page ref fields for clipping from local pdf files

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#98] [Q&A]
https://multicolumn-krystal.tiddlyhost.com/#pdfSnip

I have been using this configuration for clipping from local pdf files loaded in the browser. This was created based on this discussion - https://github.com/buggyj/tiddlyclip-plugin/discussions/73

 Is it possible to use the relative path of the pdf file in the  page ref field. The purpose is for viewing such clippings in mobile also.

**Comment by buggyj** [comment]
If you file is in a subdir of you tw you can try

`$location=((*@sublink(@pageRef)*))`

**Reply by arunnbabu81** [reply]
> 

I tried this code

![image](https://user-images.githubusercontent.com/67494083/175824420-7efc7f6f-202b-480a-a7cb-7bc566741677.png)

![image](https://user-images.githubusercontent.com/67494083/175824440-0e82f2dc-cc8d-4ffa-84cc-5ae3bf716c07.png)

![image](https://user-images.githubusercontent.com/67494083/175824461-bfb5cdd3-0b96-4b99-977b-18d2732758c5.png)

This is the file directory 

![image](https://user-images.githubusercontent.com/67494083/175824554-1c024eb5-9d3f-4909-b0c7-f81fc7cc819f.png)

![image](https://user-images.githubusercontent.com/67494083/175824566-1ef4e277-f72d-4b11-b024-dc448d502203.png)


This is what I got after snip

![image](https://user-images.githubusercontent.com/67494083/175824654-f899be00-40e1-45c7-8121-572a9b15178f.png)


Its showing the full path instead of relative path

**Comment by buggyj** [comment]
try with this - use `((*@sublinkplus(@pageRef)*))`

[substringplus.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/8990168/substringplus.json.zip)

**Reply by arunnbabu81** [reply]
> try with this - use `((*@sublinkplus(@pageRef)*))`
> 
> [substringplus.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/8990168/substringplus.json.zip)

I added the substringplus json file. Still its not working.


![image](https://user-images.githubusercontent.com/67494083/176009769-42acebf0-6743-41c9-9e87-24846160b6e8.png)


![image](https://user-images.githubusercontent.com/67494083/176009679-09188561-f795-4ceb-92d1-7fa1d1a97294.png)


![image](https://user-images.githubusercontent.com/67494083/176009945-ed508cd5-9621-4c9b-ae52-803b7b5be1a2.png)

**Comment by buggyj** [comment]
In the debug console there will be printed a message with the paths. It starts "path tohere is: " can you copy this message and paste here.

**Reply by arunnbabu81** [reply]
> In the debug console there will be printed a message with the paths. It starts "path tohere is: " can you copy this message and paste here.

![image](https://user-images.githubusercontent.com/67494083/176132650-0032ca64-5b00-40e3-8437-4f6c5c253e81.png)

**Reply by buggyj** [reply]
try with this
[substringplus.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/9008341/substringplus.json.zip)

**Reply by arunnbabu81** [reply]
> try with this [substringplus.json.zip](https://github.com/buggyj/tiddlyclip-plugin/files/9008341/substringplus.json.zip)

Thank you @buggyj .This one seem to work with windows and macbook. Will this work on mobile devices also ? 
Another related question - How do you access the snaps saved locally using save media add on in your mobile devices? Do you have such a workflow? Will those snaps load when the TW file is accessed from mobile.

**Reply by buggyj** [reply]
I don't know - I don't use my phone much apart from messages and calls.
